File: src/main/java/me/piperswe/demogui/RootComponent.java
package me.piperswe.demogui;

import javafx.geometry.Pos;
import javafx.scene.layout.VBox;
import lombok.NonNull;
import me.piperswe.utils.gui.StateContainer;
import me.piperswe.utils.gui.StaticComponent;

public class RootComponent extends StaticComponent<State> {
    public RootComponent(@NonNull StateContainer<State> stateContainer) {
        super(stateContainer);
    }

    @Override
    public VBox render() {
        var label = new LabelComponent(getStateContainer());
        var button = new ButtonComponent(getStateContainer());
        var box = new VBox(label.mount(), button.mount());
        box.setAlignment(Pos.CENTER);
        return box;
    }
}

File: src/main/kotlin/me/piperswe/demogui/kotlin/RootComponent.kt
package me.piperswe.demogui.kotlin

import javafx.geometry.Pos
import javafx.scene.layout.VBox
import me.piperswe.utils.gui.createStaticComponentFactory

val rootFactory = createStaticComponentFactory<State> { sc ->
    val label = labelFactory.create(sc)
    val button = buttonFactory.create(sc)
    val box = VBox(label.mount(), button.mount())
    box.alignment = Pos.CENTER
    box
}